### PR TITLE
public-api/viz split 2/8: chart templates for new viz linkerd-metrics-api pod

### PR DESCRIPTION
**Pull request by alpeb**
This is based on the branch `alpeb/api-move` from #5553

The underlying go code supporting the service will be pushed in a followup PR, so tests are not expected to pass on this one.

### CODE REVIEW

**Review by adleong — COMMENTED** [review]
This looks good to me.  I assume the reason this can't pass CI on its own is because the metrics-API docker image doesn't exist yet?

**Comment by alpeb** [comment]
> This looks good to me. I assume the reason this can't pass CI on its own is because the metrics-API docker image doesn't exist yet?

Yeah, lots of missing pieces at this point. The only CI runs that make sense are the ones from the last item #5560

File: viz/charts/linkerd-viz/templates/metrics-api.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+metadata:
+  annotations:
+    {{.Values.createdByAnnotation}}: {{default (printf "linkerd/helm %s" .Values.linkerdVersion) .Values.cliVersion}}
+  labels:
+    {{.Values.extensionAnnotation}}: linkerd-viz
+    app.kubernetes.io/name: metrics-api
+    app.kubernetes.io/part-of: Linkerd
+    app.kubernetes.io/version: {{default .Values.linkerdVersion .Values.controllerImageVersion}}
+    component: metrics-api
+    namespace: {{.Values.namespace}}
```
Comment (kleimkuhler) [inline viz/charts/linkerd-viz/templates/metrics-api.yaml:37]: Is the `namespace` label used anywhere? I know a few of the Viz components have this on resources, but I'm not totally sure on the purpose it serves.

This is related to the general cleanup of labels we should do across extensions anyways so not a blocker, but just curious seeing it being added here.

File: viz/charts/linkerd-viz/templates/metrics-api.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  name: linkerd-metrics-api
+  namespace: {{.Values.namespace}}
+spec:
+  replicas: {{.Values.metricsAPI.replicas}}
+  selector:
+    matchLabels:
+      {{.Values.extensionAnnotation}}: linkerd-viz
+      component: metrics-api
+      namespace: {{.Values.namespace}}
```
Comment (kleimkuhler) [inline viz/charts/linkerd-viz/templates/metrics-api.yaml:46]: And if `namespace` is removed I think selecting by extension label and component name should be enough.

File: viz/charts/linkerd-viz/templates/metrics-api.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      - args:
+        - -controller-namespace={{.Values.linkerdNamespace}}
+        - -log-level={{.Values.metricsAPI.logLevel}}
+        - -cluster-domain={{.Values.clusterDomain}}
+        {{- if .Values.prometheusUrl }}
+        - -prometheus-url={{.Values.prometheusUrl}}
+        {{- else }}
+        - -prometheus-url=http://linkerd-prometheus.linkerd-viz.svc.{{.Values.clusterDomain}}:9090
+        {{- end }}
+        image: {{.Values.metricsAPI.image.registry}}/{{.Values.metricsAPI.image.name}}:{{ default (default .Values.linkerdVersion .Values.controllerImageVersion) .Values.metricsAPI.image.tag}}
```
Comment (kleimkuhler) [inline viz/charts/linkerd-viz/templates/metrics-api.yaml:82]:
```suggestion
        image: {{.Values.metricsAPI.image.registry}}/{{.Values.metricsAPI.image.name}}:{{.Values.metricsAPI.image.tag}}
```

File: viz/charts/linkerd-viz/templates/metrics-api.yaml (hunk 1 above)
Comment (alpeb) [reply viz/charts/linkerd-viz/templates/metrics-api.yaml:37]: Yeah I don't see a purpose for this either, should be removed.

File: viz/charts/linkerd-viz/templates/metrics-api.yaml (hunk 2 above)
Comment (alpeb) [reply viz/charts/linkerd-viz/templates/metrics-api.yaml:46]: Agreed.

**Review by kleimkuhler — APPROVED** [review]
👍

**Comment by alpeb** [comment]
Closing in favor of #5560 which will aggregate all these changes.

**Pull request closed without merging**